src/operations/StandardOperation.cpp

```diff
@@ -467,4 +467,94 @@ void StandardOperation::dumpOpenQASMTeleportation(
   of << "teleport " << qreg[targets[0]].second << ", "
      << qreg[targets[1]].second << ", " << qreg[targets[2]].second << ";\n";
 }
+
+void StandardOperation::invert() {
+  switch (type) {
+  // self-inverting gates
+  case I:
+  case X:
+  case Y:
+  case Z:
+  case H:
+  case SWAP:
+  case ECR:
+    break;
+  // gates where we just update parameters
+  case GPhase:
+  case Phase:
+  case RX:
+  case RY:
+  case RZ:
+  case RXX:
+  case RYY:
+  case RZZ:
+  case RZX:
+    parameter[0] = -parameter[0];
+    break;
+  case U2:
+    std::swap(parameter[0], parameter[1]);
+    parameter[0] = -parameter[0] + PI;
+    parameter[1] = -parameter[1] - PI;
+    break;
+  case U3:
+    parameter[0] = -parameter[0];
+    parameter[1] = -parameter[1];
+    parameter[2] = -parameter[2];
+    std::swap(parameter[1], parameter[2]);
+    break;
+  case DCX:
+    std::swap(parameter[0], parameter[1]);
+    break;
+  case XXminusYY:
+  case XXplusYY:
+    parameter[0] = -parameter[0];
+    break;
+  // gates where we have specialized inverted operation types
+  case S:
+    type = Sdag;
+    break;
+  case Sdag:
+    type = S;
+    break;
+  case T:
+    type = Tdag;
+    break;
+  case Tdag:
+    type = T;
+    break;
+  case V:
+    type = Vdag;
+    break;
+  case Vdag:
+    type = V;
+    break;
+  case SX:
+    type = SXdag;
+    break;
+  case SXdag:
+    type = SX;
+    break;
+  case Peres:
+    type = Peresdag;
+    break;
+  case Peresdag:
+    type = Peres;
+    break;
+  case None:
+  case Barrier:
+  case iSWAP:
```

Comment:
As for the iSWAP gate, we currently lack the appropriate inverse type.
For now, I believe this is fine, as adding such a type would probably be out of scope for this PR.
But this is kind of a show stopper for getting rid of the code duplication in the `dd/Operations.hpp` file.
I see two ways forward:
- try to cram the new type of gate into this PR
- create a separate tracking issue for this problem and maybe add a comment here.